src/liballoc/rc.rs

```diff
@@ -2118,14 +2123,7 @@ unsafe fn data_offset<T: ?Sized>(ptr: *const T) -> isize {
     // Because it is ?Sized, it will always be the last field in memory.
```

Comment:
This is preexisting, but the `unsafe fn` should document what the requirements put upon the caller are. Here it seems that `ptr` must have valid metadata so that the alignment can be determined?

Reply:
Addressed. It's correct to say that the safety requirement is the same as `align_of_val_raw` ("plausible" metadata), but I also provided an easier, tighter overview here that encompasses all reasonable uses of this function.